${CMAKE_THREAD_LIBS_INIT} not lnked for "test/socket"

I tried compiling on ubuntu 14.04 with linuxbrew and ran into a linker error. [Here are detailed logs](https://gist.github.com/anonymous/e04f889f285cd02d8ad2) of the commands, of particular interest are the following lines:

```

cd /tmp/azmq20151012-78042-1a178n2/test/socket && /home/hbr/.linuxbrew/Cellar/cmake/3.3.2/bin/cmake -E cmake_link_script CMakeFiles/test_socket.dir/link.txt --verbose=1
/home/hbr/.linuxbrew/bin/g++-5   -Os -w -pipe -march=core2  --std=c++11   -L/home/hbr/.linuxbrew/lib -Wl,--dynamic-linker=/home/hbr/.linuxbrew/opt/glibc/lib/ld-linux-x86-64.so.2 -Wl,-rpath,/home/hbr/.linuxbrew/lib CMakeFiles/test_socket.dir/main.cpp.o  -o test_socket -rdynamic -lboost_system-mt -lboost_log-mt -lboost_date_time-mt -lboost_thread-mt -lboost_chrono-mt -lboost_regex-mt -lboost_random-mt -lzmq 
/home/hbr/.linuxbrew/bin/ld: CMakeFiles/test_socket.dir/main.cpp.o: undefined reference to symbol 'pthread_create@@GLIBC_2.2.5'
/home/hbr/.linuxbrew/lib/libpthread.so.0: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
test/socket/CMakeFiles/test_socket.dir/build.make:105: recipe for target 'test/socket/test_socket' failed
make[2]: *** [test/socket/test_socket] Error 1
make[2]: Leaving directory '/tmp/azmq20151012-78042-1a178n2'
CMakeFiles/Makefile2:286: recipe for target 'test/socket/CMakeFiles/test_socket.dir/all' failed
make[1]: *** [test/socket/CMakeFiles/test_socket.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
```

These stackoverflow questions shed some light on the issue:

- [Strange linking error: DSO missing from command line](https://stackoverflow.com/questions/19901934/strange-linking-error-dso-missing-from-command-line)
- [DSO missing from command line](https://stackoverflow.com/questions/24096807/dso-missing-from-command-line)

Apparently a library which should be linked (pthread, glibc or both) is not being linked at the moment.

pthread and pthread_create is definitely found:
```

-- Looking for include file pthread.h
-- Looking for include file pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE  
```

`-lpthread` is also linked for other tests:

```

cd /tmp/azmq20151012-78042-1a178n2/doc/examples/actor && /home/hbr/.linuxbrew/Cellar/cmake/3.3.2/bin/cmake -E cmake_link_script CMakeFiles/actor.dir/link.txt --verbose=1
/home/hbr/.linuxbrew/bin/g++-5   -Os -w -pipe -march=core2  --std=c++11   -L/home/hbr/.linuxbrew/lib -Wl,--dynamic-linker=/home/hbr/.linuxbrew/opt/glibc/lib/ld-linux-x86-64.so.2 -Wl,-rpath,/home/hbr/.linuxbrew/lib CMakeFiles/actor.dir/main.cpp.o  -o actor -rdynamic -lpthread -lzmq -lboost_system-mt -lboost_log-mt -lboost_date_time-mt -lboost_thread-mt -lboost_chrono-mt -lboost_regex-mt -lboost_random-mt 
make[2]: Leaving directory '/tmp/azmq20151012-78042-1a178n2'
```

This leads me to believe it is a problem in the cmake script for `test/socket`

--- a/test/socket/CMakeLists.txt
+++ b/test/socket/CMakeLists.txt
@@ -3,6 +3,7 @@ project(test_socket)
 add_executable(${PROJECT_NAME} main.cpp)
 
 target_link_libraries(${PROJECT_NAME} ${Boost_LIBRARIES}
-                                      ${ZeroMQ_LIBRARIES})
+                                      ${ZeroMQ_LIBRARIES}
+                                      ${CMAKE_THREAD_LIBS_INIT})
 
 add_catch_test(${PROJECT_NAME})

